src/box/iproto.cc

```diff
@@ -1043,7 +1043,7 @@ iproto_connection_input_buffer(struct iproto_connection *con)
 	 * (in only has unparsed content).
 	 */
 	if (ibuf_used(old_ibuf) == con->parse_size) {
-		ibuf_reserve_xc(old_ibuf, to_read);
+		xibuf_reserve(old_ibuf, to_read);
```

Comment:
I'm not sure it's safe to assume that allocation from IPROTO buffers never fail - technically the buffers may have quota set. OTOH if we set the quota, it'll be difficult to test all the fail paths so it maybe much better to account memory consumed by buffers in one place, at the IPROTO level, not at the buffer level. Anyway, this change looks incomplete: most of the alloc paths in the IPROTO code still handle OOM. I'd probably replace `_xc` version with explicit `diag_raise` in this patch and (maybe) remove all OOM checks from IPROTO later, in a separate commit with a proper justification. Up to @alyapunov.

Reply:
I think that use of x* versions on allocator functions should not be goal. In some cases we can think and decide that it's OK to panic in some subsystem, but there must be a practical reason for that. But we should not replace all malloc with xmalloc, for example, having the only goal to remove malloc.
I think that generally it's OK to use xregion_alloc for fiber and txn region; in all other cases we should make an identical change - replace xc version to simple, check result and raise in case of error.